imageproxy.go

```diff
@@ -266,8 +266,12 @@ func (p *Proxy) allowed(r *Request) error {
 		return nil
 	}
 
-	if len(p.SignatureKey) > 0 && validSignature(p.SignatureKey, r) {
-		return nil
+	if len(p.SignatureKey) > 0 {
```

Comment:
No need for the length check here... you can simply range over an empty or even nil slice and it will do the right thing.